bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/ColorDialog.java

```diff
@@ -184,6 +184,9 @@ public RGB open () {
 
 	/* Allocate the Custom Colors and initialize to white */
 	Display display = parent.display;
+	if(display.isRescalingAtRuntime()) {
+		OS.SetThreadDpiAwarenessContext(OS.DPI_AWARENESS_CONTEXT_UNAWARE_GDISCALED);
```

Comment:
Two remarks: 

1. I would put that code directly before the place where the dialog is opened, directly above`boolean success = OS.ChooseColor (lpcc);`
2. You need to revert the change of the DPI awareness afterwards, otherwise everything created after it, will use this awareness as well. SetThreadDpiAwarenessContext returns the previous awareness context, that you can store and reapply after the change

Same comment applies to the font dialog